routes/users.js

```diff
@@ -0,0 +1,11 @@
+const express = require('express');
+const userCtrl = require('../controllers/users');
+
+const router = express.Router();
+
+router.get('/getuser/:id?', userCtrl.getUser);
```

Comment:
Don't pass controller function directly to routes, your controller layer, should not use, `req` and `resp` objects from the routes layer, you need to decouple your code and separate logic. So use like this 

```javascript
router.get('/getuser/:id?', (req, resp)=>{
     // extract params from req
    const result  = userCtrl.getUser(someparams)
   // do something with resp
});

```

Reply:
Does it mean the controller should only return a value or throw an error? If yes, should the error details, including the status code be thrown by the controller?

Reply:
In general, the controller should not throw an error, you should have error classes, and pass status code inside them I explained this to Sergey, he has an example of that.

Yes controller should return value to router.